docs/DeveloperGuide.md

```diff
@@ -1,38 +1,201 @@
+which was part of the command.
+
+The `BuilderListComponentCommand.executeCommand()` method is then executed as part of the loop
+inside `BigPP.runLoopUntilExit()`.
+
+It first verifies:
+- That the component argument is not empty. If it is then it throws a `BuilderMissingComponentException()`
+- That the component given by the user exists in the database. If it doesn't then it throws a
+  `BuilderMissingComponentException()`
+
+It then uses the `stringToComponentListMap` HashMap from data storage to get the ComponentList associated
+with the component string input by the user (e.g. `cpu`). The `stringToComponentListMap` HashMap is initialized and
+filled when the program is first run using `DataStorage.initStringToComponentListMap()` and maps component names
+(e.g. `cpu`) to their corresponding `ComponentList` object (e.g. an object of type `CPUList`).
+
+The `ComponentList` class inherits from `ArrayList` but additionally implements the `getListString()` method which
+returns a formatted list of all the components in the `ComponentList` object.
+
+The `ComponentList` objects for each component are instantiated and populated in the `DataStorage.loadAll` method
+by loading the data in `resources/COMPONENT_NAME.json` using the `GSON` library.
+
+The program then checks if the user provided any flags to filter the output of the component command (e.g. by name,
+brand, price, etc.). If they did, then the program filters out the components not meeting the provided flags and
+creates a `ComponentList` containing all the components which do meet the provided flags.
+
+The program then uses the `ComponentList.getListString()` method to get a formatted list of the components
+of the desired type, which it then outputs to the user.
+
+A UML sequence diagram showing the interactions between the different objects involved in handling this command can be
+found below:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![List Component Command](./uml-pictures/listComponentCommand.png)
 
 
-## Product scope
-### Target user profile
+## Appendix: Requirements
+### Product scope
+#### Target user profile
 
-{Describe the target user profile}
+People who want to build PC's and keep track of their builds.
 
-### Value proposition
+#### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+ - Saves time and effort in selecting PC components
+ - Helps users avoid compatibility issues or mistakes
+ - Enables users to build cost-effective systems.
+ - Prebuilt PCs
 
-## User Stories
+### User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
```

Comment:
Is this a leftover from the template Developer Guide? There is no mention of any to-do list functionality in the rest of the guide.